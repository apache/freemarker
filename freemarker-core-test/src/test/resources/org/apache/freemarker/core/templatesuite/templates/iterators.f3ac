<#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->
<html>
<head>
<title>FreeMarker: List Iterator Test</title>
</head>
<body>
<#assign list= ["one", "two", "three", "four", "five"]>
<#assign hash = {"key", list}>
<#assign  hash2 = {"value", hash}>

<p>A simple test follows:</p>

<p>${message}</p>

<p>Now iterate over a list:</p>

<#list list as item>
<p>${item}</p>
</#list>

<p>Now iterate again:</p>

<#list list as item>
<p>${item_index}. ${item}</p>
</#list>

<p>Iterate over a list in a hash:</p>

<#list hash.key as item>
<p>${item}</p>
</#list>

<#list hash.key as item>
<p>${item}</p>
</#list>

<#list hash[ "key" ] as item>
<p>${item}</p>
</#list>

<#list hash["key"] as item>
<p>${item}</p>
</#list>

<#list hash2["value"].key as key>
<p>${key}</p>
</#list>

<#list hash2.value.key as az>
<p>${az}</p>
</#list>

</body>
</html>
