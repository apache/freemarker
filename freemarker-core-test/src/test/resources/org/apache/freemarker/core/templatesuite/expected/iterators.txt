/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
<html>
<head>
<title>FreeMarker: List Iterator Test</title>
</head>
<body>

<p>A simple test follows:</p>

<p>Hello, world!</p>

<p>Now iterate over a list:</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

<p>Now iterate again:</p>

<p>0. one</p>
<p>1. two</p>
<p>2. three</p>
<p>3. four</p>
<p>4. five</p>

<p>Iterate over a list in a hash:</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

<p>one</p>
<p>two</p>
<p>three</p>
<p>four</p>
<p>five</p>

</body>
</html>
